libcudacxx/.upstream-tests/test/std/containers/sequences/array/front_back.pass.cpp

```diff
@@ -38,6 +38,7 @@ __host__ __device__ constexpr bool check_back( double val )
 
 int main(int, char**)
 {
+#ifndef TEST_COMPILER_MSVC_2017
```

Comment:
That's a lot of tests we're commenting out here in array (here and in the next file). What's the issue that msvc 2017 is having there?

Reply:
Oh this is a good one. MSVC 2017 has issues with empty array `front ` despite all the UNREACHABLE there.

The solution is simply to wait for the refactor of array, which makes this go away.

I have no idea why this only happens here

Reply:
I realized that I have ifdef'd out too many of the test. While moving the ifdef to the right location i also added a comment why it is necessary